### ISSUE TITLE
rand is incorrect for FillVectorOfUnivariate is incorrect

### ISSUE THREAD

**Issue by torfjelde** [TuringLang/DistributionsAD.jl#68]
`rand` fills the entire vector with the same sample:
```julia
julia> using DistributionsAD
[ Info: Precompiling DistributionsAD [ced4e74d-a319-5a8a-b0ac-84af2272839c]

julia> dist = filldist(DistributionsAD.Gamma(1., 1.), 2)
Distributions.Product{Distributions.Continuous,Distributions.Gamma{Float64},FillArrays.Fill{Distributions.Gamma{Float64},1,Tuple{Base.OneTo{Int64}}}}(v=Distributions.Gamma{Float64}[Distributions.Gamma{Float64}(α=1.0, θ=1.0), Distributions.Gamma{Float64}(α=1.0, θ=1.0)])

julia> rand(dist)
2-element Array{Float64,1}:
 0.4873494685298349
 0.4873494685298349

julia> rand(dist)
2-element Array{Float64,1}:
 1.794513850833854
 1.794513850833854
```

**Comment by mohamed82008** [comment]
It seems that this is a Distributions.jl bug. We use `Product` to implement `VectorOfUnivariate` and `FillVectorOfUnivariate`.

**Comment by mohamed82008** [comment]
Closed via #69.
